src/cli-util.js

```diff
@@ -392,9 +324,188 @@ function formatFiles(argv, filepatterns) {
+      categoryOrder.splice(-1, 0, category);
+    }
+
+    groupedOptions[category].sort((a, b) => {
+      const isNoA = a.name.startsWith("no-");
+      const originalA = isNoA ? a.name.slice(3) : a.name;
+      const isNoB = b.name.startsWith("no-");
+      const originalB = isNoB ? b.name.slice(3) : b.name;
+
+      return originalA === originalB
+        ? // put `no-option` after `option`
+          isNoA - isNoB
+        : // treat `no-option` as `option`
+          originalA.localeCompare(originalB);
+    });
+  });
+
+  usage += categoryOrder
+    .map(category => {
+      return `${category} options:\n\n${indent(
+        groupedOptions[category]
+          .map(option => createOptionUsage(option, 25))
+          .join("\n"),
+        2
+      )}`;
+    })
+    .join("\n\n");
+
+  return usage + "\n\n";
+}
+
+function createOptionUsage(option, threshold) {
+  let header = `--${option.name}`;
+
+  if (option.alias) {
+    header += ` or -${option.alias}`;
+  }
+
+  switch (option.type) {
+    case "boolean":
+      // do nothing
+      break;
+    case "choice":
+      header += ` <${option.choices
+        .filter(choice => !choice.deprecated)
+        .map(choice => choice.value)
+        .join("|")}>`;
+      break;
+    default:
+      header += ` <${option.type}>`;
+      break;
+  }
+
+  if (header.length >= threshold) {
+    header += "\n" + " ".repeat(threshold);
+  } else {
+    header += " ".repeat(threshold - header.length);
+  }
+
+  return (
+    header + option.description.replace(/\n/g, "\n" + " ".repeat(threshold))
+  );
+}
+
+function capitalize(str) {
+  return str.replace(/^[a-z]/, char => char.toUpperCase());
+}
+
+function indent(str, spaces) {
+  return str.replace(/^/gm, " ".repeat(spaces));
+}
+
+function normalizeArgv(rawArgv, options) {
+  options = options || {};
+
+  const consoleWarn = options.warning === false ? () => {} : console.warn;
+
+  const normalized = {};
+
+  Object.keys(rawArgv).forEach(key => {
+    const rawValue = rawArgv[key];
+
+    if (key === "_") {
+      normalized[key] = rawValue;
+      return;
+    }
+
+    if (key.length === 1) {
+      // do nothing with alias
+      return;
+    }
+
+    const option = constant.detailOptions[key];
+
+    if (option === undefined) {
+      // unknown option
+      return;
+    }
+
+    const value = getValue(rawValue, option);
+
+    if (option.exception !== undefined) {
+      if (typeof option.exception === "function") {
+        if (option.exception(value)) {
+          normalized[key] = value;
+          return;
+        }
+      } else {
+        if (value === option.exception) {
+          normalized[key] = value;
+          return;
+        }
+      }
+    }
+
+    switch (option.type) {
+      case "int":
+        validator.validateIntOption(value, option);
+        normalized[key] = Number(value);
+        break;
+      case "choice":
+        validator.validateChoiceOption(value, option);
+      // eslint-disable-next-line no-fallthrough
```

Comment:
I think the standard way is to add a `// fall through` comment: https://eslint.org/docs/rules/no-fallthrough

Reply:
WOW, I haven't seen them before.